lib/src/pick_datetime.dart

```diff
@@ -151,7 +151,16 @@ extension NullableDateTimePick on Pick {
   DateTime? _parseIso8601() {
     final value = required().value;
     if (value is! String) return null;
-    return DateTime.tryParse(value);
+    final trimmedValue = value.trim();
+    final timeZone = RegExp(r'[a-zA-Z]+$').firstMatch(trimmedValue)?.group(0);
+    if (timeZone != null) {
+      final timeZoneOffset = _getTimeZoneOffset(timeZone);
+      // Remove the timezone from the string and add Z, so that it's parsed as UTC
+      final newValue =
+          '${trimmedValue.substring(0, trimmedValue.length - timeZone.length)}Z';
+      return DateTime.tryParse(newValue)?.add(timeZoneOffset);
```

Comment:
Maybe link to an issue that darts `DateTime` doesn't handle those timezone formats?

Reply:
Haven't found an open issue regarding any of the missing timezones
https://github.com/dart-lang/sdk/issues?q=datetime+parse+is%3Aopen